### DISCUSSION TITLE
Upload Errors, We need a way to retries failed Files

### DISCUSSION THREAD

**Discussion by Alihkhawaher** [simulot/immich-go#327] [General]
Hi,

First thank you for this amazing tool, with it, I can migrate to immich, it wouldn't be possible without it.

I have noticed that when the connection disconnects or maybe there is some delay with the server or for any reason there are some upload errors. The issue here is that we cannot re-try uploading failed files and we need to restart the whole process from the beginning which is a problem. I already tried uploading the photos three times and there are still errors, the number of errors is decreasing but it is time consuming.

Example of a report User 1: 

```
Input analysis:
---------------
scanned image file                      :   18437
scanned video file                      :    8145
scanned sidecar file                    :   26039
discarded file                          :      18
unsupported file                        :       0
file duplicated in the input            :     830
associated metadata file                :   26582
missing associated metadata file        :       0

Uploading:
----------
uploaded                                :   13508
upload error                            :    4058
file not selected                       :    1181
server's asset upgraded with the input  :       0
server has same asset                   :    7143
server has a better asset               :       2
```

Another report User 2:

```
Input analysis:
---------------
scanned image file                      :  137794
scanned video file                      :   21252
scanned sidecar file                    :  135504
discarded file                          :      11
unsupported file                        :       0
file duplicated in the input            :   28182
associated metadata file                :  159037
missing associated metadata file        :       9

Uploading:
----------
uploaded                                :   11215
upload error                            :    2801
file not selected                       :       0
server's asset upgraded with the input  :      33
server has same asset                   :  118830
server has a better asset               :     527
```

**Comment by simulot** [comment]
Thank you for your nice comment.

I'm assuming you try to upload a google takeout archive.

The log file should confirm the reason of the disconnection. 
I suspect a problem with android motion pictures. I have remarked that .MP files disturb the immich server, and it doesn't accept anything after that. I'm currently working on this issue.

I have noticed this report:
```
missing associated metadata file        :       9
```

This shouldn't occur. Are you process all takeout parts at the same time?

It's safe to restart the process even it's time consuming.

**Reply by Alihkhawaher** [reply]
Thank you for the fast reply and details,

I checked three logs, they have only one type of error
Log 1:  34,947 errors
Log 2: 40,550 errors
Log 3: 152,263 errors

Error message
```
..... dial tcp [::1]:80: connectex: No connection could be made because the target machine actively refused it.\n" time="2024-06-24T02:05.27 +03"
```

For the missing metadata, yes, I am processing them all at once. I am currently checking the zip files for any errors using 7zip test, just to make sure the files are not corrupted.

Looking forward for your next version.

Many thanks,

**Reply by simulot** [reply]
```
..... dial tcp [::1]:80: connectex: No connection could be made because the target machine actively refused it.\n" time="2024-06-24T02:05.27 +03"
```
Check in previous lines which file was upload before. If the server trips on it, any subsequent call will fails with a time out (a long one)... 

You you must process all parts at the time...  related files are distributed across parts... If you process one file at the time the JSON associated to the image won't be found, and therefore discarded. (at the moment)

**Reply by Alihkhawaher** [reply]
I checked and it did not help. Today I noticed there is disk issue so I am moving the vhdx to another disk. It keeps disconnecting. But the problem probably was all from the disk

**Reply by simulot** [reply]
You shouldn't store your photos on faulty disk...

Anyway way, the .MP files are known to hang the server. This will be corrected with the next immich-go version

**Reply by Alihkhawaher** [reply]
I didn't know the disk was faulty. The issue was because of a bad sector that caused WSL to hang. Now WSL drive is on an ssd disk, it should be fine but I did not try to use immich go again. 

The issue is not related to .mp files, I checked the logs and the previous file and the file that caused the error were both jpg

Thank you your interest

**Reply by simulot** [reply]
Docker is reputed to have problems with WSL... You should run the immich stack on a linux, or in a VM

**Reply by Alihkhawaher** [reply]
Thank you for the info.

I am using immich-distribution, a snap package. I did not need docker, it worked fine so far and it has auto update. It automatically upgraded immich from 1.105 to 1.106. I really liked how easy it was to setup.

My main PC is windows so I cannot use linux and I alo wanted gpu acceleration so WSL has what I needed

**Reply by Alihkhawaher** [reply]
I tried using Hyper-v with GPU Partitioning but it did not work
